File: gcj_2017/QR_C.pl
#!/usr/bin/env perl

# Google Code Jam 2017 Qualification Round
# Problem C. Bathroom Stalls

use bigint;

sub solve {
  my $t = $_[0], $n = int($_[1]), $k = int($_[2]);
  my $r = $n, $d = 1;
  while ($k > $d) {
    $k -= $d;
    $r -= $d;
    $d *= 2;
  }
  my $sz = int($r / $d) + ($k <= ($r % $d) ? 1 : 0) - 1;
  my $p = int($sz / 2);
  my $q = $p + ($sz % 2);
  print "Case #$t: $q $p\n";
}

my $T = <>;
for (my $t = 1; $t <= $T; $t++) {
  my ($n, $k) = split " ", <>;
  solve($t, $n, $k);
}

File: gcj_2017/QR_C.go
// Google Code Jam 2017 Qualification Round
// Problem C. Bathroom Stalls

package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

var sc = bufio.NewScanner(os.Stdin)

func nextString() string {
	sc.Scan()
	return sc.Text()
}

func nextInt() int {
	i, e := strconv.Atoi(nextString())
	if e != nil {
		panic(e)
	}
	return i
}

func solve(n int, k int) {
	r := n
	d := 1
	for k > d {
		k -= d
		r -= d
		d *= 2
	}
	e := k <= (r % d)
	sz := r / d - 1
	if e {
		sz += 1
	}
	fmt.Print(sz / 2 + (sz % 2))
	fmt.Print(" ")
	fmt.Print(sz / 2)
}

func main() {
	sc.Split(bufio.ScanWords)
	T := nextInt()
	for t := 1; t <= T; t++ {
		n := nextInt()
		k := nextInt()
		fmt.Print("Case #")
		fmt.Print(t)
		fmt.Print(": ")
		solve(n, k)
		fmt.Println("")
	}
}
